var/spack/repos/builtin/packages/fluidnumerics-self/package.py

```diff
@@ -0,0 +1,63 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class FluidnumericsSelf(CMakePackage):
+    """Spectral Element Library in Fortran."""
+
+    homepage = "https://self.fluidnumerics.com"
+    url = "https://github.com/FluidNumerics/SELF/archive/refs/tags/v0.0.1.tar.gz"
+
+    maintainers("fluidnumerics_joe", "garrettbyrd")
+
+    license("BSD-3-Clause-Attribution", checked_by="fluidnumerics_joe")
+
+    version("0.0.1", sha256="c3d4ab7d12f18b9d5864c5f9d7a91c86ebc3b7ff0fec4fa56b5c34b325344259")
+
+    variant(
+        "build_type",
+        default="Release",
+        description="The build type to build",
+        values=("Debug", "Release", "Coverage", "Profile"),
+    )
+    variant(
+        "fp64", default=True, description="Enables double precision floating point arithmetic."
+    )
+    variant(
+        "multithreading",
+        default=False,
+        description="Enables multithreading of do concurrent loops",
+    )
+    variant(
+        "nthreads",
+        default="4",
+        description="Sets the number of threads for do concurrent loops",
+        values=("2", "4", "6", "8"),
+        multi=False
+    )
```

Comment:
As an improvement for later PR, you can make this one conditional on `+multithreading`